Source/src/WixSharp.UI/ManagedUI/Forms/FeaturesDialog.cs

```diff
@@ -11,7 +9,7 @@ namespace WixSharp.UI.Forms
     /// different user experience as it has checkboxes bound to the features instead of icons context menu
     /// as MSI dialog has.
     /// </summary>
-    public partial class FeaturesDialog : ManagedForm, IManagedDialog // change ManagedForm->Form if you want to show it in designer
+    public partial class FeaturesDialog : ManagedForm // change ManagedForm->Form if you want to show it in designer
```

Comment:
Presumably, In this case, the interface acts as documentation, since ManagedForm implements IManagedDialog. When you say "hint" does do you mean code completion hints or something else. Could the hint in fact just be a comment?

Reply:
Not exactly, not code completion but a "documentation by code itself."

When as a user I see class
```
class FastList : ListBase
```
I know nothing about this class behavier unless I navigate through the whole inheritance chain one by one.

But when I see the same class declared like this:
When as a user I see class
```
class FastList : ListBase, IList, ICollection, IArrayList
```
I know what to expect from this class. Even though `ListBase` and its base classes implement these interfaces. Hiding the interfaces does not improve anything. Showing these interfaces here does not break anything neither. SQ suggestion here is nothing more than a **suggestion** in this case.

Do you agree?

Reply:
Kind of an odd one, I wouldn't expect a class to  be declared:

`class Bermese : Cat, Carnivora, Mammal, Eukaryote, LivingOrganisim, Object`

Reply:
Yes it is odd because you are referring to the changed use-case.  I provided a sample where the interfaces are independent _behavior_ interfaces. 

In your case it is an interfaces represent an inheritance chain. Quite a difference. 

You are probably not entirely accurate with "I would not expect" 😄  
When I put the interface there I was mimicking extremely informative way the VS does "Go to definition" for external assemblies:

![image](https://user-images.githubusercontent.com/16729806/81880711-7ad98280-95d1-11ea-8e5f-ea99f5a4a964.png)

So, for you, it should not be completely unexpected. 

Imagine if this MS `List` type happens to implement all interfaces in the base class. Then you, as a user, would completely loose the intend information:

```C#
public class List<T> : IList<T>, ICollection<T>, IEnumerable<T>, IEnumerable, IList, ICollection, IReadOnlyList<T>, IReadOnlyCollection<T>{}
// vs
public class List<T> : ListBase {}
``` 
We are probably in agreement on which one is more informative.

Of course it is a matter of taste, but for me, if it does not break anything and yet helps with something, it deserves to live.

Reply:
The difference is that List<T> doesn't inherit from a base class.
https://referencesource.microsoft.com/#mscorlib/system/collections/generic/list.cs,36

>We are probably in agreement on which one is more informative.
I am just trying to understand why SQ flags this as an error.

Reply:
@GilesBathgate it flags it as a code smell "'ManagedForm' implements 'IManagedDialog' so 'IManagedDialog' can be removed from the inheritance list."

sonar qube says 

"An inheritance list entry is redundant if:

It is Object - all classes extend Object implicitly.
It is int for an enum
It is a base class of another listed inheritance.
Such redundant declarations should be removed because they needlessly clutter the code and can be confusing."

Reply:
we don't have to do what it says though obviously, I for one have argued this.

Reply:
> I am just trying to understand why SQ flags this as an error.

It's not an error. SQ simply says it's not necessary. And I agree. It's not necessarily but... rather beneficial for readiability. Of course this can be challenged :) 

It's not a _code smell_ neither. 
"A code smell is a surface indication that usually corresponds to a deeper problem in the system." 

I would strongly argue that by explicitly indicating the implemented interfaces no "deeper problem" was introduced.

Reply:
@oleg-shilo I agree, I have re-introduced the interfaces :) Thank you